source/usage-examples/findOne.txt

```diff
@@ -3,3 +3,61 @@ Find a Document
 ===============
 
 .. default-domain:: mongodb
+
+You can retrieve a single document from your collection with the
+``collection.findOne()`` method. 
+
+The ``findOne()`` method takes a **context**, a **filter**, and a
+pointer to a configuration object as arguments.
+
+A context is a value of type ``context.Context``. Contexts are useful in
+concurrent server programming. A discussion of the context pattern and the
+``context`` package is outside the scope of this page, but if you are interested
+you can find more information
+`in this article on The Go Blog <https://blog.golang.org/context>`__.
+
+Your ``filter`` argument is your query to MongoDB. The ``filter`` parameter takes a query
+document that specifies the subset of your collection that you would like to
+retrieve. If you provide an empty document as your ``filter`` argument, MongoDB
+returns all the documents in your collection. The ``filter`` parameter has an empty
+interface type, but the value you pass as your ``filter`` argument must specify
+a document. For more information on documents in the MongoDB Go driver, see our
+:doc:`guide on data formats </fundamentals/data-formats>`.
+
+Your configuration object is used to specify any additional configuration
+options for your ``findOne()`` method call. A full list of supported options
+can be found in 
+:go-api:`the API documentation for the FindOneOptions type </mongo/options#FindOneOptions>`. 
+In the following example, we specify a **sort** and a **projection** using a
+value of type ``*FindOneOptions``.  
+
+Example
+-------
+
+The following code examples finds a single document from the ``movies``
+collection. It uses the following functions, methods, and values:
+
+- A **query filter** value that is passed to the ``findOne()`` method. The filter
+  matches only movies with the title exactly matching the text ``'The Room'``.
+
+- A **configuration** that **sorts** matched documents in descending order by
+  rating, such that the returned document has the highest rating. Our
+  configuration also **projects** our results so that only the ``title`` and
+  ``imdb`` fields are retrieved.
+
+- An empty **context** returned from the ``context.TODO()`` function.
```

Comment:
I think this section could be confusing to the user due to how it presents information about various parts of the example and the data on which it operates. Perhaps it could be presented more clearly in one or more of the following ways:
- provide info sequentially so the reader can follow along in the example
- describe the code without trying to identify and define the terms
- limit the description to only the most important points